vm/src/builtins/type.rs

```diff
@@ -523,6 +523,57 @@ impl PyType {
         Ok(())
     }
 
+    #[pygetset(magic)]
+    fn annotations(&self, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        if !self.slots.flags.has_feature(PyTypeFlags::HEAPTYPE) {
+            return Err(vm.new_attribute_error(format!(
+                "type object '{}' has no attribute '__annotations__'",
+                self.name()
+            )));
+        }
+
+        Ok(self
+            .attributes
+            .read()
+            .get(identifier!(vm, __annotations__))
+            .cloned()
+            .unwrap_or_else(|| {
+                self.attributes
+                    .write()
+                    .insert(identifier!(vm, __annotations__), vm.ctx.new_dict().into())
+                    .unwrap_or_else(|| vm.ctx.new_dict().into())
```

Comment:
The result of `insert` is not the newly inserted one but the old removed one.
The result of insert must be always `None` here.